### PR TITLE
Add tests for to_excel and read_excel (closes #12 and #13)

### LINKED ISSUE #12: add unit tests for Excel I/O
This is a very important part and it is not tested at all currently, and I manage to break it every other release.

### PR DESCRIPTION
**Pull request by alixdamman**

### CODE REVIEW

File: larray/tests/test_la.py [hunk 1]
```diff
@@ -2906,6 +2932,19 @@ def test_to_csv(self):
...
+        self.assertEqual(la.ndim, 5)
+        self.assertEqual(la.shape, (2, 5, 2, 2, 3))
+        self.assertEqual(la.axes.names, ['arr', 'age', 'sex', 'nat', 'time'])
+        assert_array_equal(la[x.arr[1], 0, 'F', x.nat[1], :],
+                           [3722, 3395, 3347])
+
+        la.to_excel('out.xlsx', '5d')
+        out = read_excel('out.xlsx', '5d').i[:2]
+        result = la.i[:2]
+        assert_array_equal(out, result)
```
Comment (gdementen) [inline larray/tests/test_la.py]: Since we read the full array anyway, why not compare the full array?

File: larray/tests/test_la.py [hunk 2]
```diff
@@ -2855,6 +2860,59 @@ def test_readcsv(self):
         assert_array_equal(la[x.arr[1], 0, 'F', x.nat[1], :],
                            [3722, 3395, 3347])
 
+    def test_read_excel_xlwings(self):
+        if xw is not None:
```
Comment (gdementen) [inline larray/tests/test_la.py]: We should rather use something like:
```python
    @unittest.skipIf(xw is None, "xlwings is not available")
    def test_read_excel_xlwings(self):
        pass
```
See https://docs.python.org/2/library/unittest.html#skipping-tests-and-expected-failures

File: larray/tests/test_la.py [hunk 3]
```diff
@@ -2855,6 +2860,59 @@ def test_readcsv(self):
...
+        self.assertEqual(la.axes.names, ['time'])
+        assert_array_equal(la, [3722, 3395, 3347])
+
+        la = read_excel(abspath('test.xlsx'), '2d', nb_index=1, engine='xlrd')
+        self.assertEqual(la.ndim, 2)
+        self.assertEqual(la.shape, (5, 3))
+        self.assertEqual(la.axes.names, ['age', 'time'])
+        assert_array_equal(la[0, :], [3722, 3395, 3347])
+
+        la = read_excel(abspath('test.xlsx'), '3d', index_col=[0, 1], engine=None)
```
Comment (gdementen) [inline larray/tests/test_la.py]: this will test the xlwings code path, not pandas

File: larray/tests/test_la.py [hunk 4]
```diff
@@ -2906,6 +2964,18 @@ def test_to_csv(self):
         with open('test_out1d.csv') as f:
             self.assertEqual(f.readlines(), result)
 
+    def test_to_excel(self):
+        la = read_excel(abspath('test.xlsx'), '5d', nb_index=4, engine=None)
```
Comment (gdementen) [inline larray/tests/test_la.py]: why the explicit engine=None?

File: larray/tests/test_la.py [hunk 5]
```diff
@@ -2855,6 +2860,59 @@ def test_readcsv(self):
...
+
+            la = read_excel(abspath('test.xlsx'), '5d')
+            self.assertEqual(la.ndim, 5)
+            self.assertEqual(la.shape, (2, 5, 2, 2, 3))
+            self.assertEqual(la.axes.names, ['arr', 'age', 'sex', 'nat', 'time'])
+            assert_array_equal(la[x.arr[1], 0, 'F', x.nat[1], :],
+                               [3722, 3395, 3347])
+
+    def test_read_excel_pandas(self):
+        la = read_excel(abspath('test.xlsx'), '1d', index_col=[0], engine='xlrd')
```
Comment (gdementen) [inline larray/tests/test_la.py]: it's strange that it works... there should not be any index in this case?!

Comment (alixdamman) [inline larray/tests/test_la.py]: If we don't use index_col=[0] (i.e. index_col=None), we get
{0} | time |   2007 |   2010 |   2013
    |  nan | 3722.0 | 3395.0 | 3347.0

File: larray/tests/test_la.py (hunk 3 above)
Comment (alixdamman) [inline larray/tests/test_la.py]: From read_excel function, I understand that xlwings is used if engine='xlwings'.

File: larray/tests/test_la.py (hunk 4 above)
Comment (alixdamman) [inline larray/tests/test_la.py]: Because read_excel from pandas accept either xlrd or None.

File: larray/tests/test_la.py (hunk 3 above)
Comment (gdementen) [inline larray/tests/test_la.py]: oups, sorry. This is not the same as in to_excel. We should probably have the same behavior regarding engine in both. Unsure which, but we need to be consistent.

File: larray/core.py [hunk 6]
```diff
@@ -7808,20 +7808,50 @@ def read_hdf(filepath, key, na=np.nan, sort_rows=False, sort_columns=False,
...
     """
     Reads excel file from sheet name and returns an LArray with the contents
-        nb_index: number of leading index columns (e.g. 4)
-    or
-        index_col: list of columns for the index (e.g. [0, 1, 3])
+
+    Parameters
+    ----------
+    filepath : str
+        Path where the Excel file has to be written.
```
Comment (gdementen) [inline larray/core.py]: sorry I did not see it earlier but... written should be read.

**Comment by gdementen** [comment]
LGTM, Thanks!

**Pull request merged**